packages/expo-keep-awake/src/index.ts

```diff
@@ -1,20 +1,41 @@
 import { useEffect } from 'react';
 
-import ExpoKeepAwake from './ExpoKeepAwake';
+import ExpoKeepAwake, { isAvailable } from './ExpoKeepAwake';
+import { KeepAwakeEventType } from './ExpoKeepAwake.types';
 
 const ExpoKeepAwakeTag = 'ExpoKeepAwakeDefaultTag';
 
-export function useKeepAwake(tag: string = ExpoKeepAwakeTag): void {
+function useKeepAwake(tag: string = ExpoKeepAwakeTag): void {
   useEffect(() => {
     activateKeepAwake(tag);
-    return () => deactivateKeepAwake(tag);
+    return () => {
+      deactivateKeepAwake(tag);
+    };
   }, [tag]);
 }
 
-export function activateKeepAwake(tag: string = ExpoKeepAwakeTag): void {
-  if (ExpoKeepAwake.activate) ExpoKeepAwake.activate(tag);
+async function activateKeepAwake(tag: string = ExpoKeepAwakeTag): Promise<void> {
+  if (isAvailable() && ExpoKeepAwake.activate) {
+    await ExpoKeepAwake.activate(tag);
+  }
 }
 
-export function deactivateKeepAwake(tag: string = ExpoKeepAwakeTag): void {
-  if (ExpoKeepAwake.deactivate) ExpoKeepAwake.deactivate(tag);
+async function deactivateKeepAwake(tag: string = ExpoKeepAwakeTag): Promise<void> {
+  if (isAvailable() && ExpoKeepAwake.deactivate) {
+    await ExpoKeepAwake.deactivate(tag);
+  }
 }
+
+function addEventListener(tag: string, listener: (eventType: KeepAwakeEventType) => void) {
+  if (isAvailable() && ExpoKeepAwake.addListener) {
+    ExpoKeepAwake.addListener(tag, listener);
+  }
+}
+
+export {
+  addEventListener,
+  activateKeepAwake,
+  deactivateKeepAwake,
+  useKeepAwake,
+  KeepAwakeEventType,
+};
```

Comment:
Try to keep the export pattern similar to the other modules https://github.com/expo/expo/blob/master/packages/expo-battery/src/Battery.ts

All functions should be exported inline, all types can be exported at the end of the file.